packages/wrangler/src/__tests__/pipelines.test.ts

```diff
@@ -0,0 +1,438 @@
+			    \\"type\\": \\"json\\",
+			    \\"batch\\": {},
+			    \\"compression\\": {
+			      \\"type\\": \\"gzip\\"
+			    },
+			    \\"format\\": \\"json\\",
+			    \\"path\\": {
+			      \\"bucket\\": \\"bucket\\"
+			    }
+			  },
+			  \\"endpoint\\": \\"https://0001.pipelines.cloudflarestorage.com\\"
+			}"
+		`);
+		expect(requests.count).toEqual(1);
+	});
+
+	const samplePipeline = {
+		currentVersion: 1,
+		id: "0001",
+		name: "my-pipeline",
+		metadata: {},
+		source: [
+			{
+				type: "binding",
+				format: "json",
+			},
+		],
+		transforms: [],
+		destination: {
+			type: "json",
+			batch: {},
+			compression: {
+				type: "none",
+			},
+			format: "json",
+			path: {
+				bucket: "bucket",
+			},
+		},
+		endpoint: "https://0001.pipelines.cloudflarestorage.com",
+	};
+
+	it("update - should update a pipeline", async () => {
+		const pipeline: Pipeline = samplePipeline;
+		mockShowRequest(pipeline);
+
+		const update = JSON.parse(JSON.stringify(pipeline));
+		update.destination.compression.type = "gzip";
+		const updateReq = mockUpdateRequest(update);
+
+		await runWrangler("pipelines update my-pipeline --compression gzip");
+		expect(updateReq.count).toEqual(1);
+	});
+
+	it("update - should update a pipeline with new bucket", async () => {
+		const pipeline: Pipeline = samplePipeline;
+		const tokenReq = mockCreateR2Token("new-bucket");
+		mockShowRequest(pipeline);
+
+		const update = JSON.parse(JSON.stringify(pipeline));
+		update.destination.path.bucket = "new_bucket";
+		update.destination.credentials = {
+			endpoint: "https://some-account-id.r2.cloudflarestorage.com",
+			access_key_id: "service-token-id",
+			secret_access_key:
+				"be22cbae9c1585c7b61a92fdb75afd10babd535fb9b317f90ac9a9ca896d02d7",
+		};
+		const updateReq = mockUpdateRequest(update);
+
+		await runWrangler("pipelines update my-pipeline --r2 new-bucket");
+
+		expect(tokenReq.count).toEqual(3);
+		expect(updateReq.count).toEqual(1);
+	});
+
+	it("update - should update a pipeline with new credential", async () => {
+		const pipeline: Pipeline = samplePipeline;
+		mockShowRequest(pipeline);
+
+		const update = JSON.parse(JSON.stringify(pipeline));
+		update.destination.path.bucket = "new-bucket";
+		update.destination.credentials = {
+			endpoint: "https://some-account-id.r2.cloudflarestorage.com",
+			access_key_id: "new-key",
+			secret_access_key: "new-secret",
+		};
+		const updateReq = mockUpdateRequest(update);
+
+		await runWrangler(
+			"pipelines update my-pipeline --r2 new-bucket --access-key-id new-key --secret-access-key new-secret"
+		);
+
+		expect(updateReq.count).toEqual(1);
+	});
+
+	it("delete - should delete pipeline", async () => {
+		const requests = mockDeleteRequest("foo");
+		await runWrangler("pipelines delete foo");
+
+		expect(std.err).toMatchInlineSnapshot(`""`);
+		expect(std.out).toMatchInlineSnapshot(`
+			"Deleting pipeline foo.
+			Deleted pipeline foo."
+		`);
+		expect(requests.count).toEqual(1);
+	});
+});
```

Comment:
We could add unhappy path tests here too. Things like:
- try to update a pipeline that doesn't exist
- delete a pipeline that doesn't exist
- user has invalid credentials
- bucket doesn't exist
- any other reason why create/update/delete operations might fail